views/js/previewer/provider/item/item.js

```diff
@@ -318,6 +338,8 @@ define([
             if (areaBroker) {
                 areaBroker.getToolbox().destroy();
             }
+
+            restoreContext();
```

Comment:
I see an issue with that. The context will be erased each time an item is rendered. That means the replaced context might be set as a previous context at some point. The `restoreContext()` should be called as soon as possible.

However, I don't see the point switching context. For what purpose is it made?

Reply:
After https://oat-sa.atlassian.net/browse/TAO-8134 Preview was moved from iframe to the same document as Authoring, so we need to change context for correct select elements. 
And now in the document, we have 2 objects for the same selector https://github.com/oat-sa/tao-item-runner-qti-fe/blob/ae727951e307661e0e03342c475239cb1ce7f5b7/src/qtiCommonRenderer/helpers/container.js#L32-L43
So for correct work, we need to set context https://github.com/oat-sa/tao-item-runner-qti-fe/blob/ae727951e307661e0e03342c475239cb1ce7f5b7/src/qtiCommonRenderer/helpers/container.js#L71
If we don't change context, we select the first container that is in the `Editor` not in `Preview` https://github.com/oat-sa/tao-item-runner-qti-fe/blob/ae727951e307661e0e03342c475239cb1ce7f5b7/src/qtiCommonRenderer/renderers/Object.js#L28

Reply:
Ok, fine. However, I still have a concern regarding the asymetry of the push/pop.
Anyway, it can be merged.